spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -178,35 +159,49 @@ def __get_synaptic_matrices(self, post_vertex_slice):
         matrices = SynapticMatrices(
             post_vertex_slice, self.__n_synapse_types,
             self.__all_single_syn_sz, self.__synapse_io,
-            self._synaptic_matrix_region, self._direct_matrix_region,
-            self._pop_table_region)
+            synaptic_matrix_region, direct_matrix_region,
+            pop_table_region)
```

Comment:
It seems quite dangerous to have this cache along with this change.  The SynapticMatrices object is actually a per-machine-vertex level object already.  Would it instead make sense to store this in the appropriate machine vertex?

Reply:
fair cop. will remove the creating of the cache during generate sdram (which is the second of dsg, and get sdram which do the caching). I think we need to cache them so that clear_connection_cache can work correctly.